actions/class.ItemContent.php

```diff
@@ -175,40 +150,29 @@ public function upload()
                 }
             }
 
-            $this->returnJson($filedata);
-            return;
-        } catch (PermissionException $e) {
-            $message = $e->getMessage();
-        } catch (FileUploadException $e) {
-            $message = $e->getMessage();
+            $formatter->withBody($fileData);
+        } catch (PermissionException | FileUploadException $e) {
+            $formatter->withBody(['error' => $e->getMessage()]);
         } catch (common_Exception $e) {
             $this->logWarning($e->getMessage());
-            $message = _('Unable to upload file');
+            $formatter->withBody(['error' => _('Unable to upload file')]);
         }
-        $this->returnJson(['error' => $message]);
+
+        $this->setResponse($formatter->format($this->getPsrResponse()));
```

Comment:
Lets ask @AntonNeviarouski  to test item content upload too, cause this part is very critical :)